shared/src/lib.rs

```diff
@@ -50,6 +51,21 @@ extern crate shrinkwraprs;
 #[macro_use]
 extern crate derive_new;
 
+type PravegaNodeUriParseResult<T> = std::result::Result<T, PravegaNodeUriParseError>;
+
+#[derive(Debug, Snafu)]
+pub enum PravegaNodeUriParseError {
+    #[snafu(display("Could not parse uri due to {}", error_msg))]
+    ParseError { error_msg: String },
+}
+
+#[derive(Debug, PartialEq, Default)]
```

Comment:
Do we actually want 'default'? I don't see how it does much good to construct one that's all `none`.